addons/manager.go

```diff
@@ -99,29 +100,32 @@ func (o *AddonAgentOptions) AddFlags(cmd *cobra.Command) {
 	flags.StringVar(&o.HubKubeconfigFile, "hub-kubeconfig", o.HubKubeconfigFile, "Location of kubeconfig file to connect to hub cluster.")
 	flags.StringVar(&o.SpokeClusterName, "cluster-name", o.SpokeClusterName, "Name of spoke cluster.")
 	flags.StringVar(&o.DRMode, "mode", o.DRMode, "The DR mode of token exchange addon. Valid values are: 'sync', 'async'")
-	gfs := flag.NewFlagSet("", flag.ExitOnError)
-	o.ZapOpts = zap.Options{
-		Development: true,
-	}
-	o.ZapOpts.BindFlags(gfs)
-	flags.AddGoFlagSet(gfs)
+	flags.BoolVar(&o.DevMode, "dev", false, "Set to true for dev environment (Text logging)")
 }
 
 // RunAgent starts the controllers on agent to process work from hub.
 func (o *AddonAgentOptions) RunAgent(ctx context.Context) {
-	klog.Infof("Starting addon agents.")
-
+	ctrl.SetLogger(zap.New(zap.UseDevMode(true)))
+	logger := utils.GetLogger(o.DevMode)
```

Comment:
Didn't you want to remove dependency on zap? You can configure output format without  it.